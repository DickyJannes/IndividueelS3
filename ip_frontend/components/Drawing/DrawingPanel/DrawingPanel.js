/* eslint-disable react/prop-types */
import React, {useRef} from "react";
import { View, Button } from "react-native";
import { exportComponentAsPNG } from "react-component-export-image";

import drawingStyles from "./drawingPanelStyle";
import Row from "../Row/Row";

export default function DrawingPanel(props) {
	const {width, height, selectedColor, newBackground} = props;
	const panelRef = useRef();
	let rows = [];

	for(let i = 0; i < height; i++){
		rows.push(<Row key={i} width={width} selectedColor={selectedColor} newBackground={newBackground}/>);
	}
    

	return (
		<View style={drawingStyles.drawingPanel}>
			<View style={drawingStyles.pixels} ref={panelRef}>
				{rows}
			</View>
			<Button onPress={() => exportComponentAsPNG(panelRef)} title="Export as png" />
		</View>
	);
}