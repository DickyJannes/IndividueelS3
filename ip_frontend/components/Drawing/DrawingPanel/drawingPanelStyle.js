import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
	drawingPanel: {
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
		marginVertical: 10
	},
	pixels: {
		marginBottom: "2rem"
	}
});

export default styles;