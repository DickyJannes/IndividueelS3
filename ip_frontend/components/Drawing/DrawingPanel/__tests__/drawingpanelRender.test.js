import React from 'react';
import renderer from 'react-test-renderer';

import DrawingPanel from '../DrawingPanel';

test('renders drawingpanel correctly', () => {
  const tree = renderer.create(<DrawingPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});