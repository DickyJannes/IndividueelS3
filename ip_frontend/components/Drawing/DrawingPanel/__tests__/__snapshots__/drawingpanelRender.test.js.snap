// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders drawingpanel correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": "2rem",
      }
    }
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Export as png
      </Text>
    </View>
  </View>
</View>
`;
